event_listener/dfusion_db/snapp_event_receiver.py

```diff
@@ -1,58 +1,17 @@
-from abc import abstractmethod
-
-from django_eth_events.chainevents import AbstractEventReceiver
 from .event_pusher import post_deposit, post_transition, update_accounts, initialize_accounts
+from abc import ABC, abstractmethod
 from typing import Dict, Any
 
 import logging
-
 logger = logging.getLogger(__name__)
 
-
-class GenericEventReceiver(AbstractEventReceiver):
-    name = None
-
-    def ensure_name(self, _name):
-        return self.name == _name
-
-    def parse_event(self, decoded_event):
-        res = {param['name']: param['value'] for param in decoded_event['params']}
-
-        # Convert byte strings to hex
-        for k, v in res.items():
-            if isinstance(v, bytes):
-                res[k] = v.hex()
-
-        logging.info("{} received {}".format(self.name, res))
-        return res
-
-    def save(self, decoded_event, block_info=None):
-        if not self.ensure_name(decoded_event['name']):
-            return
-
-        parsed_event = self.parse_event(decoded_event)
-
-        self.real_save(parsed_event, block_info)
-
-    def rollback(self, decoded_event, block_info=None):
-        if not self.ensure_name(decoded_event['name']):
-            return
-
-        self.real_rollback(decoded_event, block_info)
-
-    @abstractmethod
-    def real_save(self, decoded_event, block_info=None):
-        pass
-
+class SnappEventListener(ABC):
+    """Abstract SnappEventReceiver class."""
     @abstractmethod
-    def real_rollback(self, decoded_event, block_info=None):
-        pass
+    def save(self, event:Dict[str, Any], block_info): pass
```

Comment:
Should we also include rollback here?

Reply:
Right now, there is no functionality for rollbacks (it's a TODO to add it). Once we add this functionality, we will have to add a method there, yes.

Reply:
Good enough for me!